pkg/operator/options/options_validation.go

```diff
@@ -83,6 +84,13 @@ func (o Options) validateVMMemoryOverheadPercent() error {
 	return nil
 }
 
+func (o Options) validateProvisionMode() error {
+	if o.ProvisionMode != "" && o.ProvisionMode != "bootstrappingclient" {
+		return fmt.Errorf("provision-mode is invalid: %s", o.ProvisionMode)
+	}
+	return nil
+}
```

Comment:
again think that just putting this behind the managed karpenter flag makes the most sense, unless you plan on having more modes.

Reply:
(discussed offline + in a different thread) This is intentional and we keep this.